File: hyunjae/프로그래머스/고득점키트/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {};
        //제곱근까지 순회
        for(int i=1; i<=(int) Math.sqrt(yellow); i++){
            //정수의 곱으로 나타낼 수 있으면
            if(yellow%i ==0){
                int temp = yellow/i;
                //총 크기를 구하고 노란색만큼 때면 갈색 갯수
                int total = (i+2)*(temp+2) - yellow;
                if(total == brown) {
                    //갈색전체갯수가 주어진 갈색 갯수와 같으면 
                    answer = new int[] {temp+2, i+2};
                    break;
                }
            }
        }
        return answer;
    }
}

File: sejeong/카펫.py
def solution(brown,yellow):
    area = brown + yellow  # 전체격자를 더해 넓이를 계산
    arr = []
    for i in range(1,area+1):  # 두수를 곱해서 넓이가 나올수있는 쌍 구하기
        if area % i == 0:
            if i >= area //i:  # 가로가 크거나 같으므로
                arr.append((i,area//i))  # (가로,세로) 쌍 넣기
    for a,b in arr:  # 가로세로가 될수있는 경우를 돌면서 그경우의 갈색격자수의 개수가 가지고있는격자수와일치하는지
        if (a * 2) + ((b-2) *2) == brown:
            return [a,b]
brown = 10
yellow = 2
print(solution(brown,yellow))
# 카펫의 가로 세로 크기 출력 , 가로는 세로와 같거나 더 길다